src/main/java/com/project/trackfit/service/impl/UserServiceImplementation.java

```diff
@@ -0,0 +1,27 @@
+package com.project.trackfit.service.impl;
+
+import com.project.trackfit.model.User;
+import com.project.trackfit.repository.UserRepository;
+import com.project.trackfit.service.UserService;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+@AllArgsConstructor
+public class UserServiceImplementation implements UserService {
+
+    private UserRepository userRepository;
+
+    @Override
+    public User createUser(User user) {
+        return userRepository.save(user);
+    }
+
+    @Override
+    public User getUserById(Long userId) {
+        Optional<User> optionalUser = userRepository.findById(userId);
+        return optionalUser.get();
```

Comment:
this will fail if user does not exits.

You want to throw an exeception if user does not exits

Reply:
Replaced the get method with optionalUser.orElseThrow(). From documentation about orElseThrow( ) -> "If a value is present, returns the value, otherwise throws NoSuchElementException." Is that enough or should we add a try / catch block?